h2o-algos/src/main/java/hex/tree/TreeHandler.java

```diff
@@ -62,10 +63,106 @@ else if (!(possibleModel instanceof SharedTreeModel) && !(possibleModel instance
+    }
+
+    private static String getNodeRepresentation(SharedTreeNode node, String languageRepresentation, int padding) {
+        if (node.getLeftChild() != null || node.getRightChild() != null) {
+            if (padding != 0)
+                languageRepresentation += getNewPaddedLine(padding);
+            languageRepresentation += getConditionLine(node);
+            if (node.getRightChild() != null) {
+                languageRepresentation += getNewPaddedLine(padding);
+                languageRepresentation = getNodeRepresentation(node.getRightChild(), languageRepresentation, padding + 1);
+            }
+            languageRepresentation += getNewPaddedLine(padding);
+            languageRepresentation += getElseLine(node);
+            if (node.getLeftChild() != null) {
+                languageRepresentation += getNewPaddedLine(padding);
+                languageRepresentation = getNodeRepresentation(node.getLeftChild(), languageRepresentation, padding + 1);
+            }
+            languageRepresentation += getNewPaddedLine(padding);
+            languageRepresentation += "}";
+        } else {
+            if (Float.compare(node.getPredValue(),Float.NaN) != 0) {
+                languageRepresentation += getNewPaddedLine(padding);
+                languageRepresentation += "Predicted value: " + node.getPredValue();
+                languageRepresentation += getNewPaddedLine(padding);
+            }
+        }
+        return languageRepresentation;
+    }
+
+    private static String getNewPaddedLine(int padding) {
+        String line = "\n";
+        for(int i = 0; i < padding; i++) {
+            line += "\t";
+        }
+        return line;
+    }
+
+    private static String getElseLine(SharedTreeNode node) {
+        StringBuilder elseLine = new StringBuilder();
+        if (Float.compare(node.getSplitValue(),Float.NaN) != 0) {
+            elseLine.append("} else {");
+        } else {
+            SharedTreeNode leftChild = node.getLeftChild();
+            elseLine.append("} else if ( ").append(node.getColName()).append(" is in [ ");
+            String stringToParseInclusiveLevelsFrom = leftChild.getInclusiveLevels().toString();
+            int inclusiveLevelsLength = leftChild.getInclusiveLevels().toString().length();
+            if (inclusiveLevelsLength  > 2) {
+                // get rid of curly braces:
+                stringToParseInclusiveLevelsFrom = stringToParseInclusiveLevelsFrom.substring(1, inclusiveLevelsLength - 1);
+                String[] inclusiveLevels = stringToParseInclusiveLevelsFrom.split(",");
+                for (String index : inclusiveLevels) {
+                    elseLine.append(node.getDomainValues()[Integer.parseInt(index.trim())] + " ");
+                }
+            } else {
+                elseLine.append("Missing set of levels for underlying node");
+            }
+            elseLine.append("]) {");
+        }
+        return elseLine.toString();
```

Comment:
why not return the builder and append it to parrent builder?

Reply:
fixed